crates/ra_hir/src/attr_macros.rs

```diff
@@ -0,0 +1,13 @@
+use ra_syntax::ast::{self, AstNode};
+
+mod derive;
+
+pub fn expand_attr_macro(attr_node: ast::Attr, target_node: ast::ModuleItem) -> Option<tt::Subtree> {
+    if !derive::is_derive_attr(&attr_node) {
+        log::warn!("Unimplemented macro attr: {}", attr_node.syntax().to_string());
+
+        return None;
+    } else {
+        derive::expand_derive_attr(attr_node, target_node)
+    }
```

Comment:
it would make more sense to me to write this the other way around:
```rust
if derive::is_derive_attr(&attr_node) {
    return derive::expand_derive_attr(attr_node, target_node);
}
log::warn!("Unimplemented macro attr: {}", attr_node.syntax().to_string());
None
```
this way, the `is_derive_attr` check is right next to the `expand_derive_attr` call, and it's easier to add more new attributes